File: app/src/main/java/ru/startandroid/develop/retrofittask/MyRecyclerViewAdapter.java
package ru.startandroid.develop.retrofittask;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class MyRecyclerViewAdapter extends RecyclerView.Adapter<MyRecyclerViewAdapter.MyViewHolder> {

    private List<MessageEntity> messageList;
    private Context context;
    private Callback callback;

    interface Callback{
        void onItemLongClicked(MessageEntity message);
    }

    public MyRecyclerViewAdapter(Context context){
        this.context = context;
    }

    public void setItems(List<MessageEntity> messageList){
        this.messageList = messageList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item, parent, false);
        MyViewHolder myViewHolder = new MyViewHolder(view, callback);
        return myViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        MessageEntity message = messageList.get(position);
        holder.bind(message);
    }

    @Override
    public int getItemCount() {
        if(messageList == null)
            return  0;
        else {
            return messageList.size();
        }
    }

    static class MyViewHolder extends RecyclerView.ViewHolder{

        private TextView id;
        private TextView userId;
        private TextView titleTextView;
        private TextView messageTextView;
        private Callback callback;

        public MyViewHolder(@NonNull View itemView, Callback callback) {
            super(itemView);
            id = itemView.findViewById(R.id.id);
            userId = itemView.findViewById(R.id.user_Id);
            titleTextView = itemView.findViewById(R.id.titleTextView);
            messageTextView = itemView.findViewById(R.id.messageTextView);
            this.callback = callback;
        }

        public void bind(final MessageEntity message){
            id.setText("Id: " + Integer.toString(message.getId()));
            userId.setText("UserId: " + Integer.toString(message.getUserId()));
            titleTextView.setText("Title: " + message.getTitle());
            messageTextView.setText(message.getText());

            itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View view) {
                    callback.onItemLongClicked(message);
                    return false;
                }
            });

        }
    }
}

File: app/src/main/java/ru/startandroid/develop/retrofittask/MyRecyclerViewAdapter.kt
package ru.startandroid.develop.retrofittask

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import ru.startandroid.develop.retrofittask.MyRecyclerViewAdapter.MyViewHolder

class MyRecyclerViewAdapter(private val context: Context) : RecyclerView.Adapter<MyViewHolder>() {
    private var messageList: List<MessageEntity>? = null
    private val callback: Callback? = null

    interface Callback {
        fun onItemLongClicked(message: MessageEntity?)
    }

    fun setItems(messageList: List<MessageEntity>?) {
        this.messageList = messageList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item, parent, false)
        return MyViewHolder(view, callback)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val message = messageList!![position]
        holder.bind(message)
    }

    override fun getItemCount(): Int {
        return if (messageList == null) 0 else {
            messageList!!.size
        }
    }

    class MyViewHolder internal constructor(itemView: View, callback: Callback?) : ViewHolder(itemView) {
        val id: TextView = itemView.findViewById(R.id.id)
        val userId: TextView = itemView.findViewById(R.id.user_Id)
        val titleTextView: TextView = itemView.findViewById(R.id.titleTextView)
        val messageTextView: TextView = itemView.findViewById(R.id.titleTextView)
        val callback: Callback? = callback
        fun bind(message: MessageEntity) {
            id.text = "Id: " + Integer.toString(message.id)
            userId.text = "UserId: " + Integer.toString(message.userId)
            titleTextView.text = "Title: " + message.title
            messageTextView.text = message.text
            itemView.setOnLongClickListener {
                callback!!.onItemLongClicked(message)
                false
            }
        }

    }
}